pandas/core/frame.py

```diff
@@ -637,8 +637,30 @@ class DataFrame(NDFrame, OpsMixin):
     def _constructor(self) -> Callable[..., DataFrame]:
         return DataFrame
 
+    def _constructor_from_mgr(self, mgr, axes):
+        if self._constructor is DataFrame:
+            # we are pandas.DataFrame (or a subclass that doesn't override _constructor)
+            return self._from_mgr(mgr, axes=axes)
+        else:
+            assert axes is mgr.axes
+            return self._constructor(mgr)
+
     _constructor_sliced: Callable[..., Series] = Series
 
+    def _sliced_from_mgr(self, mgr, axes) -> Series:
+        # https://github.com/pandas-dev/pandas/pull/52132#issuecomment-1481491828
+        #  This is a short-term implementation that will be replaced
+        #  with self._constructor_sliced._from_mgr(...)
+        #  once downstream packages (geopandas) have had a chance to implement
+        #  their own overrides.
+        return self._constructor_sliced(mgr)
```

Comment:
Shouldn't this be something like `Series._from_mgr(mgr)` or `self._constructor_sliced._from_mgr(mgr)` ? 
Because now this is calling the same as the fall-back in `_constructor_sliced_from_mgr`, and thus the `if` block there is not doing anything different.